chat-plugins/health-fitness.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Heath and Fitness chat-plugin
+ * Pokemon Showdown - http://pokemonshowdown.com/
+ * 
+ * This handles health and fitness daily challenges as well as a daily quote
+ * 
+ * @license MIT license
+ */
+'use strict';
+
+const FS = require('../fs');
+
+const HF_DATA_PATH = 'config/chat-plugins/health-and-fitness.json';
+const MAX_DATA_LENGTH = 500;
+
+class HealthFitness {
+	constructor() {
+		this.hfData = {
+			cardio: '',
+			gym: '',
+			quote: '',
+			__proto__: null,
+		};
+		try {
+			this.hfData = require(`../${HF_DATA_PATH}`);
+		} catch (error) {
+			if (error.code !== 'MODULE_NOT_FOUND') throw error;
+		}
+
+		this.saveData = (() => {
+			let writing = false;
+			let writePending = false;
+			return async () => {
+				if (writing) {
+					writePending = true;
+					return;
+				}
+				writing = true;
+
+				await FS(`${HF_DATA_PATH}.0`).write(JSON.stringify(this.hfData));
+				await FS(`${HF_DATA_PATH}.0`).rename(HF_DATA_PATH);
+
+				writing = false;
+				if (writePending) {
+					writePending = false;
+					setImmediate(() => this.saveData());
+				}
+			};
+		})();
+	}
+	setCardio(cardio) {
+		this.hfData.cardio = cardio;
+		this.saveData();
+	}
+	setGym(gym) {
+		this.hfData.gym = gym;
+		this.saveData();
+	}
+	setQuote(quote) {
+		this.hfData.quote = quote;
+		this.saveData();
+	}
+	getData() {
```

Comment:
This could be setup like
```javascript
get data() {
return this.hfData;
}
```
So that `HealthFitness#data` dosent need to be called as a function, it can just be called like a variable.

Reply:
Ugh no do not do this ever

Reply:
Getters and setters should only be used when there's a specific reason why pretending they're variables is a good idea.

Reply:
Although, like, why can't we read `.hfData` directly? Is there a need for this sort of indirection?

Reply:
Good point @Zarel